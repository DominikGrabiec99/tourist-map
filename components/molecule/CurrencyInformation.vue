<template>
  <div class="currency flex pl-3 items-center">
    <div
      class="flag-image text-2xl pr-3"
      :style="`--background-position-x-var: ${currency.bgPositionX}; --background-position-y-var: ${currency.bgPositionY};`"
    />
    <div class="flex flex-col">
      <span class="text-xs">Kurs {{ currency.symbol }}: </span>
      <span class="text-xs">{{ currency?.mid?.toFixed(2) }} z≈Ç</span>
    </div>
  </div>
</template>

<script setup lang="ts">
/** INTERFACES */
import IMainCurrency from '@/ts/interfaces/MainCurrency';

defineProps<{
  currency: IMainCurrency;
}>();
</script>

<style scoped lang="scss">
.currency {
  min-width: 130px;
  .flag-image {
    &:after {
      content: '';
      background-position: var(--background-position-x-var)
        var(--background-position-y-var);
      background-image: url('/images/all_flags.svg');
      border-radius: 50%;
      width: 1em;
      height: 1em;
      transform: translateZ(0);
      display: block;
      background-size: 68.57143em 68.57143em;
      box-shadow: 0 0 0 2px $white;
    }
  }
}
</style>
