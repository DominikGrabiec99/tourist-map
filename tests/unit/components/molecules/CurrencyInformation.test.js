import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import CurrencyInformation from '@/components/molecule/CurrencyInformation.vue';

const currency = {
  code: 'USD',
  symbol: '$',
  bgPositionX: '-19.32145em',
  bgPositionY: '-3.25em',
  name: 'dolar',
  mid: 4.1191,
  priority: 1,
};

describe('CurrencyInformation.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(CurrencyInformation, {
      props: {
        currency,
      },
    });
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render props currency', () => {
    expect(wrapper.text()).toContain(currency.mid.toFixed(2));
  });
});
