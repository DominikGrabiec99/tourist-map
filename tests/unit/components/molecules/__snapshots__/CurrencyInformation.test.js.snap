// Vitest Snapshot v1

exports[`CurrencyInformation.vue > render component 1`] = `
<div
  class="currency flex pl-3 items-center"
  data-v-5a1df612=""
>
  <div
    class="flag-image text-2xl pr-3"
    data-v-5a1df612=""
    style="--background-position-x-var: -19.32145em; --background-position-y-var: -3.25em;"
  />
  <div
    class="flex flex-col"
    data-v-5a1df612=""
  >
    <span
      class="text-xs"
      data-v-5a1df612=""
    >
      Kurs $: 
    </span>
    <span
      class="text-xs"
      data-v-5a1df612=""
    >
      4.12 zł
    </span>
  </div>
</div>
`;

exports[`Header.vue > render component 1`] = `
<div
  class="currency flex pl-3 items-center"
  data-v-5a1df612=""
>
  <div
    class="flag-image text-2xl pr-3"
    data-v-5a1df612=""
    style="--background-position-x-var: -19.32145em; --background-position-y-var: -3.25em;"
  />
  <div
    class="flex flex-col"
    data-v-5a1df612=""
  >
    <span
      class="text-xs"
      data-v-5a1df612=""
    >
      Kurs $: 
    </span>
    <span
      class="text-xs"
      data-v-5a1df612=""
    >
      4.12 zł
    </span>
  </div>
</div>
`;
